Build failure: `llama-swap` on `x86_64-darwin` (failing test)

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

On an Intel Mac:
```shell
$ nix build 'nixpkgs#llama-swap'
```
_or_
```
nix-build '<nixpkgs>' -A llama-swap
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/307149000

### Relevant log output

```console
Using versionCheckHook
Running phase: unpackPhase
unpacking source archive /nix/store/vr3qy7b2vk7ni512fr8rrh5mach3l1h4-source
source root is source
Running phase: patchPhase
Running phase: configurePhase
Running phase: buildPhase
Building subPackage .
Building subPackage ./event
Building subPackage ./misc/simple-responder
Building subPackage ./proxy
buildPhase completed in 1 minutes 44 seconds
Running phase: checkPhase
ok      github.com/mostlygeek/llama-swap/event  1.315s
--- FAIL: TestProcess_AutomaticallyStartsUpstream (5.79s)
    process_test.go:44: 
                Error Trace:    /private/tmp/nix-build-llama-swap-156.drv-0/source/proxy/process_test.go:44
                Error:          Not equal: 
                                expected: "ready"
                                actual  : "stopped"
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,2 @@
                                -(proxy.ProcessState) (len=5) "ready"
                                +(proxy.ProcessState) (len=7) "stopped"
                                 
                Test:           TestProcess_AutomaticallyStartsUpstream
    process_test.go:46: 
                Error Trace:    /private/tmp/nix-build-llama-swap-156.drv-0/source/proxy/process_test.go:46
                Error:          Not equal: 
                                expected: 200
                                actual  : 502
                Test:           TestProcess_AutomaticallyStartsUpstream
                Messages:       Expected status code 200, got 502
    process_test.go:47: 
                Error Trace:    /private/tmp/nix-build-llama-swap-156.drv-0/source/proxy/process_test.go:47
                Error:          "unable to start process: health check timed out after 5s\n" does not contain "testing91931"
                Test:           TestProcess_AutomaticallyStartsUpstream
--- FAIL: TestProcess_StopCmd (5.30s)
    process_test.go:465: 
                Error Trace:    /private/tmp/nix-build-llama-swap-156.drv-0/source/proxy/process_test.go:465
                Error:          Expected nil, but got: &errors.errorString{s:"health check timed out after 2s"}
                Test:           TestProcess_StopCmd
    process_test.go:466: 
                Error Trace:    /private/tmp/nix-build-llama-swap-156.drv-0/source/proxy/process_test.go:466
                Error:          Not equal: 
                                expected: "stopped"
                                actual  : "ready"
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,2 @@
                                -(proxy.ProcessState) (len=7) "stopped"
                                +(proxy.ProcessState) (len=5) "ready"
                                 
                Test:           TestProcess_StopCmd
FAIL
FAIL    github.com/mostlygeek/llama-swap/proxy  52.250s
FAIL
```

### Additional context

I found out that @06kellyjac had also run into this and reported it as <https://github.com/mostlygeek/llama-swap/issues/320>. They were running into the issue on GitHub Actions, so at first I thought it was an upstream issue, but then I realized that the workflow in question was also using Nix (specifically `nixpkgs-review`) to try and build `llama-swap`.

### System metadata

 - system: `"x86_64-darwin"`
 - host os: `Darwin 24.5.0, macOS 15.5`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.28.5`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/01yzp96024m0hnj71sdnd58ic23sqcp9-nixpkgs/nixpkgs`

### Notify maintainers

@06kellyjac @podium868909 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ll/llama-swap/package.nix
+++ b/pkgs/by-name/ll/llama-swap/package.nix
@@ -16,13 +16,13 @@ let
 in
 buildGoModule (finalAttrs: {
   pname = "llama-swap";
-  version = "156";
+  version = "165";
 
   src = fetchFromGitHub {
     owner = "mostlygeek";
     repo = "llama-swap";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-z0262afVjsdGe6WPoWO1wbccLO538fXBuxOOqLnJHRU=";
+    hash = "sha256-3NlA4LnAJ1qCy1+Jcv6wrPg/7trQhpwx00Sk98V7ZdY=";
     # populate values that require us to use git. By doing this in postFetch we
     # can delete .git afterwards and maintain better reproducibility of the src.
     leaveDotGit = true;
@@ -38,11 +38,14 @@ buildGoModule (finalAttrs: {
   vendorHash = "sha256-5mmciFAGe8ZEIQvXejhYN+ocJL3wOVwevIieDuokhGU=";
 
   passthru.ui = callPackage ./ui.nix { llama-swap = finalAttrs.finalPackage; };
-  passthru.npmDepsHash = "sha256-Sbvz3oudMVf+PxOJ6s7LsDaxFwvftNc8ZW5KPpbI/cA=";
+  passthru.npmDepsHash = "sha256-F6izMZY4554M6PqPYjKcjNol3A6BZHHYA0CIcNrU5JA=";
 
   nativeBuildInputs = [
     versionCheckHook
   ];
+
+  # required for testing
+  __darwinAllowLocalNetworking = true;
 
   ldflags = [
     "-s"
@@ -70,6 +73,29 @@ buildGoModule (finalAttrs: {
     # it's unneeded
     "misc/simple-responder"
   ];
+
+  checkFlags =
+    let
+      skippedTests = lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [
+        # Fail only on x86_64-darwin intermittently
+        # https://github.com/mostlygeek/llama-swap/issues/320
+        "TestProcess_AutomaticallyStartsUpstream"
+        "TestProcess_WaitOnMultipleStarts"
+        "TestProcess_BrokenModelConfig"
+        "TestProcess_UnloadAfterTTL"
+        "TestProcess_LowTTLValue"
+        "TestProcess_HTTPRequestsHaveTimeToFinish"
+        "TestProcess_SwapState"
+        "TestProcess_ShutdownInterruptsHealthCheck"
+        "TestProcess_ExitInterruptsHealthCheck"
+        "TestProcess_ConcurrencyLimit"
+        "TestProcess_StopImmediately"
+        "TestProcess_ForceStopWithKill"
+        "TestProcess_StopCmd"
+        "TestProcess_EnvironmentSetCorrectly"
+      ];
+    in
+    [ "-skip=^${builtins.concatStringsSep "$|^" skippedTests}$" ];
 
   # some tests expect to execute `simple-something` and proxy/helpers_test.go
   # checks the file exists

